src/core/server/models/user/user.ts

```diff
@@ -671,6 +682,63 @@ export async function updateUserPassword(
   return result.value || null;
 }
 
+export async function scheduleDeletionDate(
+  mongo: Db,
+  tenantID: string,
+  userID: string,
+  deletionDate: Date
+) {
+  const users = collection(mongo);
+  const result = await users.findOneAndUpdate(
+    {
+      id: userID,
+      tenantID,
+    },
+    {
+      $set: {
+        scheduledDeletionDate: deletionDate,
+      },
+    },
+    {
+      returnOriginal: false,
+    }
+  );
+
+  if (!result.value) {
+    throw new Error("Unable to update user deletion date.");
+  }
+
+  return result.value || null;
+}
+
+export async function clearDeletionDate(
+  mongo: Db,
+  tenantID: string,
+  userID: string
+) {
+  const users = collection(mongo);
```

Comment:
Avoid creating collection variables like this, prefer inline use of `collection(mongo)`.